[draft] Nicholas.hulston/fips metrics

nhulston opened the pull request:
<!--- Please remember to review the [contribution guidelines](https://github.com/DataDog/datadog-lambda-js/blob/master/CONTRIBUTING.md) if you have not yet done so._  --->

### What does this PR do?

TODO manually test

### Motivation

<!--- What inspired you to submit this pull request? --->

### Testing Guidelines

<!--- How did you test this pull request? --->

### Additional Notes

<!--- Anything else we should know when reviewing? --->

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

apiarian-datadog commented on src/metrics/listener.spec.ts:
```diff
@@ -123,41 +123,11 @@ describe("MetricsListener", () => {
     await expect(listener.onCompleteInvocation()).resolves.toEqual(undefined);
   });
 
-  it("configures FIPS endpoint for GovCloud regions", async () => {
```
should we keep a version of this test like `configures FIPS endpoint for FIPS_MODE_ENABLED`?

nhulston replied:
I moved those tests to `fips.spec.ts`; since we're moving the FIPS enabled/disabled logic to there, this was a little redundant

apiarian-datadog reviewed (approved):
looking good. added a note about a deleted test but fine otherwise. we'll add statsd timestamp support separately?

nhulston commented:
> looking good. added a note about a deleted test but fine otherwise. we'll add statsd timestamp support separately?

Thanks! Yes, I added timestamp support in https://github.com/DataDog/datadog-lambda-js/pull/648
